controllers/helpers_test.go

```diff
@@ -229,3 +232,200 @@ func TestGetPrismCentralClientForCluster(t *testing.T) {
+				mockv3Service.EXPECT().ListAllImage(gomock.Any(), gomock.Any()).Return(
+					&prismclientv3.ImageListIntentResponse{
+						Entities: []*prismclientv3.ImageIntentResponse{
+							{
+								Spec: &prismclientv3.Image{
+									Name: ptr.To("example"),
+								},
+							},
+						},
+					}, nil,
+				)
+
+				return &prismclientv3.Client{V3: mockv3Service}
+			},
+			image: infrav1.NutanixResourceIdentifier{
+				Type: infrav1.NutanixIdentifierName,
+				Name: ptr.To("example"),
+			},
+			want: &prismclientv3.ImageIntentResponse{
+				Spec: &prismclientv3.Image{
+					Name: ptr.To("example"),
+				},
+			},
+		},
+		{
+			name: "image name matches multiple images",
+			clientBuilder: func() *prismclientv3.Client {
+				mockctrl := gomock.NewController(t)
+				mockv3Service := mocknutanixv3.NewMockService(mockctrl)
+				mockv3Service.EXPECT().ListAllImage(gomock.Any(), gomock.Any()).Return(
+					&prismclientv3.ImageListIntentResponse{
+						Entities: []*prismclientv3.ImageIntentResponse{
+							{
+								Spec: &prismclientv3.Image{
+									Name: ptr.To("example"),
+								},
+							},
+							{
+								Spec: &prismclientv3.Image{
+									Name: ptr.To("example"),
+								},
+							},
+						},
+					}, nil,
+				)
+
+				return &prismclientv3.Client{V3: mockv3Service}
+			},
+			image: infrav1.NutanixResourceIdentifier{
+				Type: infrav1.NutanixIdentifierName,
+				Name: ptr.To("example"),
+			},
+			wantErr: true,
+		},
+		{
+			name: "image name matches zero images",
+			clientBuilder: func() *prismclientv3.Client {
+				mockctrl := gomock.NewController(t)
+				mockv3Service := mocknutanixv3.NewMockService(mockctrl)
+				mockv3Service.EXPECT().ListAllImage(gomock.Any(), gomock.Any()).Return(
+					&prismclientv3.ImageListIntentResponse{
+						Entities: []*prismclientv3.ImageIntentResponse{},
+					}, nil,
+				)
+
+				return &prismclientv3.Client{V3: mockv3Service}
+			},
+			image: infrav1.NutanixResourceIdentifier{
+				Type: infrav1.NutanixIdentifierName,
+				Name: ptr.To("example"),
+			},
+			wantErr: true,
+		},
+		{
+			name: "image name matches one image",
+			clientBuilder: func() *prismclientv3.Client {
+				mockctrl := gomock.NewController(t)
+				mockv3Service := mocknutanixv3.NewMockService(mockctrl)
+				mockv3Service.EXPECT().ListAllImage(gomock.Any(), gomock.Any()).Return(
+					&prismclientv3.ImageListIntentResponse{
+						Entities: []*prismclientv3.ImageIntentResponse{
+							{
+								Spec: &prismclientv3.Image{
+									Name: ptr.To("example"),
+								},
+							},
+						},
+					}, nil,
+				)
+
+				return &prismclientv3.Client{V3: mockv3Service}
+			},
+			image: infrav1.NutanixResourceIdentifier{
+				Type: infrav1.NutanixIdentifierName,
+				Name: ptr.To("example"),
+			},
+			want: &prismclientv3.ImageIntentResponse{
+				Spec: &prismclientv3.Image{
+					Name: ptr.To("example"),
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
nit: Why here ordinary unit tests and in `nutanixmachine_controller_test.go` Ginkgo unit tests?

Reply:
Because you wrote the Ginkgo test in #493, and I adapted it! :joy: 

BTW, I can convert the Ginkgo test an ordinary (standard library) unit test, but then we might need to do the same in #493.  I prefer ordinary unit tests, so maybe I could do the converson in a later PR.

Reply:
In my opinion Gingko tests are good enough and it will be good if we'll discuss what kind of unit tests frameworks we would like to use in our projects.

Anyway, I think we need to plan a meeting for this discussion.
Feel free to resolve this conversation.